Libplanet/Blockchain/BlockChain.cs

```diff
@@ -735,7 +735,9 @@ internal HashDigest<SHA256> FindBranchPoint(BlockLocator locator)
 
                 foreach (HashDigest<SHA256> hash in locator)
                 {
-                    if (Blocks.ContainsKey(hash))
+                    if (Blocks.ContainsKey(hash)
+                        && Blocks[hash] is Block<T> block
```

Comment:
Isn't this check redundant with the above line?

Reply:
I checked the hash first because `Blocks[hash]` throws `KeyNotFoundException` if the hash doesn't exists. Or catching the exception is better?

Reply:
👌  I've misled. this line is not only a check but also binding `block`.

Reply:
I guess it's it's for avoiding catching an exception?